libnopegl/src/ngpu/bindgroup.h

```diff
@@ -40,7 +40,8 @@ enum ngpu_access {
     NGPU_ACCESS_READ_BIT   = 1 << 0,
     NGPU_ACCESS_WRITE_BIT  = 1 << 1,
     NGPU_ACCESS_READ_WRITE = NGPU_ACCESS_READ_BIT | NGPU_ACCESS_WRITE_BIT,
-    NGPU_ACCESS_NB
+    NGPU_ACCESS_NB,
+    NGPU_ACCESS_MAX_ENUM = 0x7FFFFFFF
```

Comment:
I guess you don't want to add the include, but `INT32_MAX` could have been more explicit.

Also, what prevents the compiler from picking a larger size like 64-bit enum here? I'd be more comfortable with a static assert (just one somewhere in the internal header or something) that checks that such an enum is exactly 32-bit long wide.

Reply:
Yep I didn't want to include stdint.h but we use it in the code base anyway so I'm not opposed to it.

> Also, what prevents the compiler from picking a larger size like 64-bit enum here? I'd be more comfortable with a static assert (just one somewhere in the internal header or something) that checks that such an enum is exactly 32-bit long wide.

Quoting the C11 standard (6.7.2.2):

```
The expression that defines the value of an enumeration constant shall be an integer
constant expression that has a value representable as an int.
```

I will add a static assert as you suggested.

Reply:
Static asserts added:
`api: name anonymous enums and ensure they are stored on 32-bit`:
``` diff
--- a/libnopegl/src/api.c
+++ b/libnopegl/src/api.c
@@ -54,6 +54,12 @@
 #include "vaapi_ctx.h"
 #endif
 
+// Ensure public enums are stored on 32-bit
+NGLI_STATIC_ASSERT(enum_on_32bit, sizeof(enum ngl_log_level) == sizeof(int32_t));
+NGLI_STATIC_ASSERT(enum_on_32bit, sizeof(enum ngl_platform_type) == sizeof(int32_t));
+NGLI_STATIC_ASSERT(enum_on_32bit, sizeof(enum ngl_backend_type) == sizeof(int32_t));
+NGLI_STATIC_ASSERT(enum_on_32bit, sizeof(enum ngl_capture_buffer_type) == sizeof(int32_t));
+

```